consensus/src/chained_bft/block_storage/mod.rs

```diff
@@ -57,21 +57,20 @@ pub trait BlockReader: Send + Sync {
     fn path_from_root(&self, block_id: HashValue)
         -> Option<Vec<Arc<ExecutedBlock<Self::Payload>>>>;
 
-    /// Generates and returns a block with the given parent and payload.
+    /// Generates and returns a proposal to create a new block with the given parent and payload.
     /// Note that it does not add the block to the tree, just generates it.
-    /// The main reason we want this function in the BlockStore is the fact that the signer required
-    /// for signing the newly created block is held by the block store.
+    /// The main reason we want this function in the BlockStore maintains the QuorumCert.
     /// The function panics in the following cases:
     /// * If the parent or its quorum certificate are not present in the tree,
     /// * If the given round (which is typically calculated by Pacemaker) is not greater than that
     ///   of a parent.
-    fn create_block(
+    fn create_proposal(
```

Comment:
Huh, I don't think we need it here any more. It becomes silly because the "parent" is found based on the highest QC, hence, we don't really need to first find the highest QC, then get its block, then pass the block to the block store, so that it'd take the QC again and generate the proposal.
We could leave it as is just to make this PR size reasonable but open a separate small issue to pull it out.

Reply:
Opened issue https://github.com/libra/libra/issues/1482 to pull it out

Reply:
Ahh, that makes sense! And it would be a pretty nice separation of responsibilities! It is still a shame that proposal generator needs to know about the block store. Should we consider removing that requirement as well? As far as I can see, the dependency only (really) exists so we can see if we are a round at reconfiguration, perhaps we should pass that in a as a boolean along with the highest qc?

Reply:
sadly proposal generator also needs to know the `path_from_root`